File: src/main/java/com/stage/teamb/services/EmployeeService.java
package com.stage.teamb.services;

import com.stage.teamb.dtos.AddressDTO;
import com.stage.teamb.dtos.DepartmentDTO;
import com.stage.teamb.dtos.EmployeeDTO;
import com.stage.teamb.dtos.RatingDTO;
import com.stage.teamb.models.Employee;

import java.util.List;
import java.util.Optional;

public interface EmployeeService {

    List<EmployeeDTO> findAllEmployees();

    EmployeeDTO findEmployeeById(Long id);

    EmployeeDTO saveEmployee(EmployeeDTO employeeDTO);

    void deleteEmployeeById(Long id);
    EmployeeDTO updateEmployee(EmployeeDTO employeeDTO);


    AddressDTO associateEmployeeWithAddress(Long addressId, Long employeeId);

    List<EmployeeDTO> findEmployeesByDepartmentId(Long departmentId);

    AddressDTO disassociateEmployeeFromAddress(Long addressId);

    DepartmentDTO assignDepartmentToEmployee(Long employeeId, Long departmentId);

    DepartmentDTO unassignDepartmentFromEmployee(Long employeeId);

    RatingDTO createRating(Long employeeId, Long publicationId, RatingDTO ratingDTO);

    RatingDTO updateRating(Long ratingId, RatingDTO ratingDTO);

    void deleteRating(Long ratingId);

    List<Employee> findAll();

    Optional<Employee> findOne(Long id);

    Employee saveOne(Employee employee);

    void deleteOne(Long id);

    Boolean existsById(Long id);
}

File: src/main/resources/teamb-front/src/app/shared/services/employee.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, Subscription } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {
  
  private baseUrl = 'http://localhost:8082/api/employees';


  constructor(private http: HttpClient) { }

 
  registerEmployee(request: any): Observable<any> {
    return this.http.post<any>(`${this.baseUrl}/register/employee`, request);
  }

  findAllEmployees(): Observable<any[]> {
    const request = this.http.get<any[]>(`${this.baseUrl}/find/all`);
    return request;
  }

  findEmployeeById(id: number): Observable<any> {
    const request = this.http.get<any>(`${this.baseUrl}/${id}`);
    return request;
  }

  deleteEmployeeById(id: number): Observable<void> {
    const request = this.http.delete<void>(`${this.baseUrl}/delete/${id}`);
    return request;
  }

  updateEmployee(employeeDTO: any): Observable<any> {
    const request = this.http.put<any>(`${this.baseUrl}/update/${employeeDTO.id}`, employeeDTO);
    return request;
  }

  associateEmployeeWithAddress(addressId: number, employeeId: number): Observable<any> {
    const request = this.http.post<any>(`${this.baseUrl}/${employeeId}/address/${addressId}`, {});
    return request;
  }

  disassociateEmployeeFromAddress(addressId: number): Observable<any> {
    const request = this.http.delete<any>(`${this.baseUrl}/${addressId}/disassociate`);
    return request;
  }

  findEmployeesByDepartmentId(departmentId: number): Observable<any[]> {
    const request = this.http.get<any[]>(`${this.baseUrl}/department/find/all/${departmentId}`);
    return request;
  }

  assignDepartmentToEmployee(employeeId: number, departmentId: number): Observable<any> {
    const request = this.http.post<any>(`${this.baseUrl}/${employeeId}/department/${departmentId}`, {});
    return request;
  }

  unassignDepartmentFromEmployee(employeeId: number): Observable<any> {
    const request = this.http.delete<any>(`${this.baseUrl}/${employeeId}/unassign`);
    return request;
  }

  
}
